### DISCUSSION TITLE
Capture 'click' event on a track-data

### DISCUSSION THREAD

**Discussion by miguelocana** [gosling-lang/gosling.js#1090] [Q&A] [answered]
Hi, is it possible to capture the 'click' event on a track-data element? For instance, to redirect to a URL.

![image](https://github.com/user-attachments/assets/f8fcd57b-accb-486f-9c59-64fc50d4d68f)

**Comment by sehilyi** [comment, marked as answer]
Hi @miguelocana!

You can subscribe to the "click" event through JS API ([doc](http://gosling-lang.org/docs/js-api#subscribe)).

You can find a similar example in our Gosling React Example repository, where it uses the "rangeSelect" event instead:

https://github.com/gosling-lang/gosling-react-example/blob/e11178bc255af6bb4653f9923052e326bf55b720/src/example/MouseEvents.jsx#L19-L37

Be sure to turn on the click event in your track specification, i.e.,

```ts
experimental: {
    mouseEvents: true
},
```

Example: https://github.com/gosling-lang/gosling-react-example/blob/e11178bc255af6bb4653f9923052e326bf55b720/src/example/MouseEvents.jsx#L83C14-L89C17

**Reply by miguelocana** [reply]
Thank you for replying so quickly, and sorry for the delay. I'm using Vue3, and the implementation is being a bit more tedious. By following the documentation and the examples you provided, I managed to find a workaround:

```typescript
onUpdated(async () => {
  let data = {
    ...,
    views: [
      {
        ...,
        tracks: [
          {
            ...,
            experimental: {
              mouseEvents: true
            }
          }
        ]
      }
    ]
  }
  let api = await embed(document.getElementById("gosling-container"), data)

  api.subscribe("click", (event, item) => {
    console.log(event, item)
  })

})
```
